src/java/org/apache/ivy/core/cache/DefaultRepositoryCacheManager.java

```diff
@@ -1215,26 +1217,22 @@ public ResolvedResource resolve(Artifact artifact) {
                     for (int i = 0; i < confs.length; i++) {
                         Artifact[] arts = md.getArtifacts(confs[i]);
                         for (int j = 0; j < arts.length; j++) {
-                            Artifact transformedArtifact = NameSpaceHelper.transform(
-                                arts[j], options.getNamespace().getToSystemTransformer());
-                            ArtifactOrigin origin = getSavedArtifactOrigin(
-                                transformedArtifact);
-                            File artFile = getArchiveFileInCache(
-                                transformedArtifact, origin, false);
-                            if (artFile.exists()) {
-                                Message.debug("deleting " + artFile);
-                                if (!artFile.delete()) {
-                                    // Old artifacts couldn't get deleted!
-                                    // Restore the original ivy file so the next time we
-                                    // resolve the old artifacts are deleted again
-                                    backupDownloader.restore();
-                                    Message.error("Couldn't delete outdated artifact from cache: " + artFile);
-                                    return null;
-                                }
-                            }
-                            removeSavedArtifactOrigin(transformedArtifact);
+                            if (!prepAndDeleteArtifact(arts[j], options, backupDownloader)) return null;
                         }
                     }
+
+                    final Artifact sourceArtifact = new MDArtifact(md, mrid.getName(), "src", "jar", null,
+                        Collections.singletonMap("m:classifier", "sources"));
+
+                    final Artifact srcArtifact = new MDArtifact(md, mrid.getName(), "src", "jar", null,
+                        Collections.singletonMap("m:classifier", "src"));
+
+                    final Artifact javadocArtifact = new MDArtifact(md, mrid.getName(), "doc", "jar", null,
+                        Collections.singletonMap("m:classifier", "javadoc"));
```

Comment:
AFAIK it's an sbt's understanding of ivy when code was first written vs. ivy thing.   I believe it's trouble all of our brewing, however it is ALSO the case that inside of maven repositories "-src" vs. "-sources" is a thing and you need to support both, depending on which project is pushing the sources.